pkg/types/baremetal/auth/ironic_auth.go

```diff
@@ -0,0 +1,56 @@
+package auth
+
+import (
+	"crypto/rand"
+	"math/big"
+)
+
+var inited = make(chan bool, 1)
+var ironicPassword string
+
+func init() {
+	inited <- false
+}
+
+func generateRandomPassword() (string, error) {
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 16
+	buf := make([]rune, length)
+	numChars := big.NewInt(int64(len(chars)))
+	for i := range buf {
+		c, err := rand.Int(rand.Reader, numChars)
+		if err != nil {
+			return "", err
```

Comment:
Could we wrap this error? If rand fails due to lack of entropy or something, the panic later might not be clear where things went wrong to a user.